WordPress/gutenberg
# New User Experience (NUX)

### 🕺 What this is

Closes #3670. Blocked by https://github.com/WordPress/gutenberg/pull/7038.

This implements the NUX flow outlined by @karmatosed in https://github.com/WordPress/gutenberg/issues/3670#issuecomment-384595696:

When a new user opens Gutenberg, they're presented with a guide that points out some useful page elements:

![nux](https://user-images.githubusercontent.com/612155/40633279-0d4e915c-6332-11e8-80e2-39f10f2c838e.gif)

If they dismiss the guide, it won't ever show again:

![nux-dismiss](https://user-images.githubusercontent.com/612155/40633389-ccc2127a-6332-11e8-991b-004f5f0da523.gif)

### 🤔 How it works

- A `DotTip` component renders a tip bubble independently of anything else
  - This uses `Popover`, which now supports having its `yAxis` set to `middle`
- The `triggerGuide()` action lets you associate a series of tips into a single guide
- The `isTipVisible()` and `getAssociatedGuide()` selectors return information about which step in a guide the user is in
- The `dismissTip()` and `disableTips()` actions control dismissing a single tip (and stepping through the guide) and disabling tips entirely
- All of this functionality is contained within a new `nux` module with the intention that these components should be made available in other areas of the WP Admin

### 📋 Testing

There's unit tests included. Manual testing can be done by:

1. Create a new post
2. View and step through the tips
3. Refresh the page—the current step should be maintained 
3. Dismiss the tips
4. Refresh the page—the tips should no longer appear

### ✍️ Copy

I'm still waiting on the final copy for each tip. There was some discussion about this in https://github.com/WordPress/gutenberg/issues/3670.
